src/gfx/main.cpp

```diff
@@ -0,0 +1,761 @@
+					break;
+				} else if (c == '\r') {
+					file.sbumpc(); // Discard the '\n'
+					break;
+				}
+				argPool.push_back(c);
+			}
+			argPool.push_back('\0');
+
+			// Discard whitespace until the next argument (candidate)
+			while (isblank(c)) {
+				c = file.sbumpc();
+			}
+			if (c == '\r') {
+				c = file.sbumpc(); // Skip the '\n'
+			}
+		} while (c != '\n' && c != EOF); // End if we reached EOL
+	}
+}
+/**
+ * Parses an arg vector, modifying `options` as options are read.
+ * The three booleans are for the "auto path" flags, since their processing must be deferred to the
+ * end of option parsing.
+ *
+ * Returns NULL if the vector was fully parsed, or a pointer (which is part of the arg vector) to an
+ * "at-file" path if one is encountered.
+ */
+static char *parseArgv(int argc, char **argv, bool &autoAttrmap, bool &autoTilemap,
+                       bool &autoPalettes) {
+	int opt;
+
+	while ((opt = musl_getopt_long_only(argc, argv, optstring, longopts, nullptr)) != -1) {
+		char *arg = musl_optarg; // Make a copy for scanning
+		switch (opt) {
+		case 'A':
+			autoAttrmap = true;
+			break;
+		case 'a':
+			autoAttrmap = false;
+			options.attrmap = musl_optarg;
+			break;
+		case 'b':
+			options.baseTileIDs[0] = parseNumber(arg, "Bank 0 base tile ID", 0);
+			if (options.baseTileIDs[0] >= 256) {
+				error("Bank 0 base tile ID must be below 256");
+			}
+			if (*arg == '\0') {
+				options.baseTileIDs[1] = 0;
+				break;
+			}
+			skipWhitespace(arg);
+			if (*arg != ',') {
+				error("Base tile IDs must be one or two comma-separated numbers, not \"%s\"",
+				      musl_optarg);
+				break;
+			}
+			skipWhitespace(arg);
+			options.baseTileIDs[1] = parseNumber(arg, "Bank 1 base tile ID", 0);
+			if (options.baseTileIDs[1] >= 256) {
+				error("Bank 1 base tile ID must be below 256");
+			}
+			if (*arg != '\0') {
+				error("Base tile IDs must be one or two comma-separated numbers, not \"%s\"",
+				      musl_optarg);
+				break;
+			}
+			break;
+		case 'C':
+			options.useColorCurve = true;
+			break;
+		case 'c':
+			if (musl_optarg[0] == '#') {
+				options.palSpecType = Options::EXPLICIT;
+				parseInlinePalSpec(musl_optarg);
+			} else if (strcasecmp(musl_optarg, "embedded") == 0) {
+				// Use PLTE, error out if missing
+				options.palSpecType = Options::EMBEDDED;
+			} else {
+				options.palSpecType = Options::EXPLICIT;
+				// Can't parse the file yet, as "flat" color collections need to know the palette
+				// size to be split; thus, we defer that
+				// TODO: this does not validate the `fmt` part of any external spec but the last
+				// one, but I guess that's okay
+				externalPalSpec = musl_optarg;
+			}
+			break;
+		case 'd':
+			options.bitDepth = parseNumber(arg, "Bit depth", 2);
+			if (*arg != '\0') {
+				error("Bit depth (-b) argument must be a valid number, not \"%s\"", musl_optarg);
+			} else if (options.bitDepth != -1 && options.bitDepth != 1 && options.bitDepth != 2) {
```

Comment:
```suggestion
			} else if (options.bitDepth != 1 && options.bitDepth != 2) {
```

Reply:
Ah, actually, in this context, `-1` means that the number was not parsed successfully.

Reply:
Though the error value is overridden to `2` in this case, so the check truly is useless.